src/analyze/analyze-unit-shell.c

```diff
@@ -0,0 +1,120 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "sd-bus.h"
+
+#include "alloc-util.h"
+#include "analyze.h"
+#include "analyze-unit-shell.h"
+#include "bus-error.h"
+#include "bus-util.h"
+#include "fd-util.h"
+#include "log.h"
+#include "namespace-util.h"
+#include "process-util.h"
+#include "runtime-scope.h"
+#include "strv.h"
+#include "unit-def.h"
+#include "unit-name.h"
+
+int verb_unit_shell(int argc, char *argv[], void *userdata) {
+        _cleanup_(sd_bus_flush_close_unrefp) sd_bus *bus = NULL;
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *reply = NULL;
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_free_ char *unit = NULL;
+        int r;
+
```

Comment:
could you add a check that arg_transport is BUS_TRANSPORT_LOCAL here, and fail if not? i.e. if people do "-M" or "-H" we shouldn't try to connect to any PID reported here because it's not going to make sense locally.